### PR TITLE
TATB - 30 :[ BUG ] - When click skip button , pop up will remain even after question change

### PR DESCRIPTION
**Pull request by charmijtdev**
When we skip a question then a pop up will appear asking if you ant to skip or not but after the timer gets over and the question will change but the pop up will still remain.

### CODE REVIEW

File: src/app/component/quiz/quiz.component.ts [hunk 1]
```diff
@@ -122,6 +122,9 @@ export class Quizcomponent implements OnInit, OnDestroy {
 
   next(questionindex: number): void {
     this.carousel.next();
+    if (this.dialogService.hasModelOpen()) {
+      this.dialogService.destroy();
+    }
     const values = this.FormArray.controls[questionindex].value.radioValue;
     this.answer(questionindex, values);
     this.disabledValuesAndForm();
```
Comment (PoojanPatelJT) [inline src/app/component/quiz/quiz.component.ts:130]: Herer on line 132 
his.questionindex = questionindex + 1;
    if (this.questionindex + 1 == this.question.length) {
      this.submit();
    } 
    
    

Could you please check without adding 1 as i believe that it should work as increment is done before that

**Review by PoojanPatelJT — CHANGES_REQUESTED** [review]

File: src/app/component/quiz/quiz.component.ts (hunk 1 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/quiz/quiz.component.ts:130]: yes i have removed adding 1 as it is working fine without it

Comment (PrabhatJTDeveloper) [reply src/app/component/quiz/quiz.component.ts:130]: yes i have removed adding 1 as increment is done before

File: src/app/component/quiz/quiz.component.ts [hunk 2]
```diff
@@ -219,7 +222,7 @@ export class Quizcomponent implements OnInit, OnDestroy {
         this.points = this.points -= 0.25;
         this.inCorrectanswer++;
       }
-      if (questionindex === this.question.length) {
+      if (questionindex + 1 === this.question.length) {
         this.submit();
```
Comment (PoojanPatelJT) [inline src/app/component/quiz/quiz.component.ts:223]: Please wrap condition like that  (questionindex + 1)

File: src/app/component/quiz/quiz.component.ts [hunk 3]
```diff
@@ -122,11 +122,14 @@ export class Quizcomponent implements OnInit, OnDestroy {
 
   next(questionindex: number): void {
     this.carousel.next();
+    if (this.dialogService.hasModelOpen()) {
+      this.dialogService.destroy();
+    }
     const values = this.FormArray.controls[questionindex].value.radioValue;
     this.answer(questionindex, values);
```
Comment (PoojanPatelJT) [inline src/app/component/quiz/quiz.component.ts:129]: Please use camelcase here and also add undefined check here at line no. 128

**Review by PoojanPatelJT — CHANGES_REQUESTED** [review]

File: src/app/component/quiz/quiz.component.ts [hunk 4]
```diff
@@ -219,7 +222,7 @@ export class Quizcomponent implements OnInit, OnDestroy {
         this.points = this.points -= 0.25;
         this.inCorrectanswer++;
       }
-      if (questionindex === this.question.length) {
+      if (questionindex + 1 === this.question.length) {
```
Comment (charmijtdev) [inline src/app/component/quiz/quiz.component.ts:222]: I dont think so cehcking this condition twice would be fine as checking on either answer of check on next

File: src/app/dialog-service/dialog.service.ts [hunk 5]
```diff
@@ -5,31 +5,42 @@ import { ModalComponent } from './modal/modal/modal.component';
 import dialogData from '../../assets/json/dialogData.json';
 import { Router } from '@angular/router';
 
-
 @Injectable({
   providedIn: 'root',
 })
 export class DialogService {
   adminform: any;
```
Comment (charmijtdev) [inline src/app/dialog-service/dialog.service.ts]: what is asminform is doing here ?

File: src/app/dialog-service/dialog.service.ts [hunk 6]
```diff
@@ -5,31 +5,42 @@ import { ModalComponent } from './modal/modal/modal.component';
...
       modalRef.componentInstance.yesButtonLable = yesButtonLable;
       modalRef.componentInstance.NoButtonLable = NoButtonLable;
-      
-      modalRef.result.then((data) => {
-        resolve(<boolean> data)
-      }); 
-    })
+
+      modalRef.result
+        .then((data) => {
```
Comment (charmijtdev) [inline src/app/dialog-service/dialog.service.ts]: Could you please change data to answer ?

Comment (PrabhatJTDeveloper) [reply src/app/dialog-service/dialog.service.ts]: Yes i have changed it

File: src/app/dialog-service/dialog.service.ts (hunk 5 above)
Comment (PrabhatJTDeveloper) [reply src/app/dialog-service/dialog.service.ts]: it is not in use hence removed

File: src/app/component/quiz/quiz.component.ts (hunk 2 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/quiz/quiz.component.ts:223]: This condition is already removed in branch number TATB - 35

File: src/app/component/quiz/quiz.component.ts (hunk 4 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/quiz/quiz.component.ts:222]: This condition is already removed in branch number TATB - 35

File: src/app/component/quiz/quiz.component.ts (hunk 3 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/quiz/quiz.component.ts:129]: Yes i have used camelcase and added undefined check

File: src/app/component/quiz/quiz.component.html [hunk 7]
```diff
@@ -27,7 +27,7 @@
             >
             <ng-template
               ngbSlide
-              *ngFor="let arrayloop of FormArray.controls; let i = index"
+              *ngFor="let arrayloop of formArray.controls; let i = index"
             >
               <ng-container [formGroupName]="i">
                 <div class="card" style="width: 60rem; margin-top: 7rem">
```
Comment (charmijtdev) [inline src/app/component/quiz/quiz.component.html]: Please remove this inline css

Comment (PrabhatJTDeveloper) [reply src/app/component/quiz/quiz.component.html]: i have already removed it in another PR

**Comment by charmijtdev** [comment]
OLDPR

**Pull request closed without merging**